js/ascendex.js

```diff
@@ -467,55 +483,101 @@ module.exports = class ascendex extends Exchange {
-            const type = ('useLot' in market) ? 'spot' : 'future';
+            const type = ('useLot' in market) ? 'spot' : 'swap';
             const spot = (type === 'spot');
-            const future = (type === 'future');
+            const perpetual = (type === 'swap');
             const margin = this.safeValue (market, 'marginTradable', false);
+            let contract = false;
+            if (margin || perpetual === true) {
+                contract = true;
+            }
+            let linear = false;
+            if (perpetual) {
+                linear = true;
+            }
+            let minQty = this.safeNumber (market, 'minQty');
+            let maxQty = this.safeNumber (market, 'maxQty');
+            let minPrice = this.safeNumber (market, 'tickSize');
+            let maxPrice = undefined;
             let symbol = id;
-            if (!future) {
+            if (!perpetual) {
                 symbol = base + '/' + quote;
+            } else {
+                const lotSizeFilter = this.safeValue (market, 'lotSizeFilter');
+                minQty = this.safeNumber (lotSizeFilter, 'minQty');
+                maxQty = this.safeNumber (lotSizeFilter, 'maxQty');
+                const priceFilter = this.safeValue (market, 'priceFilter');
+                minPrice = this.safeNumber (priceFilter, 'minPrice');
+                maxPrice = this.safeNumber (priceFilter, 'maxPrice');
+                const str = this.safeString (market, 'underlying');
+                const substrings = str.split ('/');
+                baseId = substrings[0];
+                quoteId = substrings[1];
+                base = this.safeCurrencyCode (baseId);
+                quote = this.safeCurrencyCode (quoteId);
+                symbol = base + '/' + quote + ':' + settle;
             }
             const fee = this.safeNumber (market, 'commissionReserveRate');
             result.push ({
                 'id': id,
                 'symbol': symbol,
                 'base': base,
                 'quote': quote,
+                'settle': settle,
                 'baseId': baseId,
                 'quoteId': quoteId,
-                'info': market,
+                'settleId': settleId,
                 'type': type,
                 'spot': spot,
                 'margin': margin,
-                'future': future,
+                'swap': perpetual,
+                'future': false,
+                'option': false,
                 'active': active,
-                'precision': precision,
+                'derivative': contract,
+                'contract': contract,
+                'linear': linear,
+                'inverse': undefined,
                 'taker': fee,
                 'maker': fee,
+                'contractSize': undefined,
+                'expiry': undefined,
+                'expiryDatetime': undefined,
+                'strike': undefined,
+                'optionType': undefined,
+                'fees': undefined,
```

Comment:
You'll probably have to do something like this here https://github.com/ccxt/ccxt/pull/10936/files#r771775958

Reply:
We should simply remove this line from here.